uspv/init.go

```diff
@@ -137,7 +120,42 @@ func (s *SPVCon) Connect(remoteNode string) error {
 		return err
 	}
 	s.WBytes += uint64(n)
+	return nil
+}
+
+// Connect dials out and connects to full nodes.
+func (s *SPVCon) Connect(remoteNode string) error {
+	var err error
+	var listOfNodes []string
+	if lnutil.YupString(remoteNode) {
+		// if remoteNode is "yes" but no IP specified, use DNS seed
+		listOfNodes, err = s.GetListOfNodes()
+		if err != nil {
+			return err
+		}
+	} else { // else connect to user-specified node
+		if !strings.Contains(remoteNode, ":") {
+			remoteNode = remoteNode + ":" + s.Param.DefaultPort
+		}
+		listOfNodes = append(listOfNodes, remoteNode)
+	}
 
+	for len(listOfNodes) != 0 {
+		err = s.DialNode(listOfNodes)
+		if err != nil {
+			listOfNodes = listOfNodes[1:]
+			continue
+		}
+		err = s.Handshake(listOfNodes)
+		if err != nil {
+			// means we either have a sapm node or didn't get a resonse. So we Try again
+			log.Println(err)
+			log.Println("Couldn't establish connection with node. Proceeding to the next one")
+			listOfNodes = listOfNodes[1:]
+			continue
+		}
+		break
```

Comment:
Need an error message for when it runs out of nodes to connect to, and still hasn't done a successful handshake.

Reply:
thanks!